### PR TITLE
[kotlin][client] Small improvements

### PR DESCRIPTION
**Pull request by WonderCsabo**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jimschubert  @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m

### CODE REVIEW

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapitools/client/models/User.kt [hunk 1]
```diff
@@ -17,6 +17,7 @@ package org.openapitools.client.models
 
 
 import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
```
Comment (WonderCsabo) [inline samples/client/petstore/kotlin/src/main/kotlin/org/openapitools/client/models/User.kt:20]: Is there a way to only add this import when an enum class is inside the data class?

File: samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/models/Pet.kt [hunk 2]
```diff
@@ -67,6 +67,7 @@ data class Pet (
      * Values: AVAILABLE,PENDING,SOLD
      */
     @KSerializable
+
```
Comment (WonderCsabo) [inline samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/models/Pet.kt:70]: Is there a way to skip this empty line, when not moshi is used as a serialisation library?

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapitools/client/models/User.kt (hunk 1 above)
Comment (4brunu) [reply samples/client/petstore/kotlin/src/main/kotlin/org/openapitools/client/models/User.kt:20]: Not that I'm aware of

File: samples/client/petstore/kotlin-retrofit2/src/main/kotlin/org/openapitools/client/apis/PetApi.kt [hunk 3]
```diff
@@ -41,7 +41,7 @@ interface PetApi {
     /**
     * enum for parameter status
     */
-    enum class Status_findPetsByStatus(val value: kotlin.String) {
+    enum class StatusFindPetsByStatus(val value: kotlin.String) {
```
Comment (4brunu) [inline samples/client/petstore/kotlin-retrofit2/src/main/kotlin/org/openapitools/client/apis/PetApi.kt:44]: @WonderCsabo here we have a breaking change

File: modules/openapi-generator/src/main/resources/kotlin-client/enum_class.mustache [hunk 4]
```diff
@@ -30,6 +31,7 @@ import kotlinx.serialization.*
  * Values: {{#allowableValues}}{{#enumVars}}{{&name}}{{^-last}},{{/-last}}{{/enumVars}}{{/allowableValues}}
  */
 {{#multiplatform}}@Serializable{{/multiplatform}}{{#kotlinx_serialization}}@Serializable{{#enumUnknownDefaultCase}}(with = {{classname}}Serializer::class){{/enumUnknownDefaultCase}}{{/kotlinx_serialization}}
+{{#moshi}}@JsonClass(generateAdapter = false){{/moshi}}
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/kotlin-client/enum_class.mustache:34]: To fix the empty space I think you need to format this line like this:
```
{{#moshi}}
@JsonClass(generateAdapter = false)
{{/moshi}}
```
I'm not 100% sure of this, but try it and see if it works.

File: samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/models/Pet.kt (hunk 2 above)
Comment (4brunu) [reply samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/models/Pet.kt:70]: I left a tip to fix this in a comment I opened.

File: modules/openapi-generator/src/main/resources/kotlin-client/enum_class.mustache (hunk 4 above)
Comment (WonderCsabo) [reply modules/openapi-generator/src/main/resources/kotlin-client/enum_class.mustache]: Thanks, it worked!

**Comment by WonderCsabo** [comment]
@4brunu i addressed all review comments.

**Comment by 4brunu** [comment]
Looks good to me 👍

**Comment by 4brunu** [comment]
@WonderCsabo Thank you for taking the time to improve openapi-generator 👍

**Pull request merged**